src/main/kotlin/com/avail/descriptor/pojos/EqualityRawPojoDescriptor.kt

```diff
@@ -0,0 +1,139 @@
+ *     list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ *  * Neither the name of the copyright holder nor the names of the contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.avail.descriptor.pojos
+
+import com.avail.annotations.AvailMethod
+import com.avail.descriptor.representation.A_BasicObject
+import com.avail.descriptor.representation.AvailObject
+import com.avail.descriptor.representation.Mutability
+import java.util.*
+
+/**
+ * `EqualityRawPojoDescriptor` differs from [RawPojoDescriptor] in that equality
+ * of wrapped pojos is based on semantic equivalence rather than referential
+ * equality. It is used only for immutable reflective classes and exists only to
+ * support certain recursive comparison operations.
+ *
+ * @constructor
+ *
+ * @param mutability
+ *   The [mutability][Mutability] of the new descriptor.
+ * @param javaObject
+ *   The actual Java [Object] represented by the [AvailObject] that will use the
+ *   new descriptor.
+ *
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ * @see RawPojoDescriptor
+ */
+internal class EqualityRawPojoDescriptor(
+	mutability: Mutability,
+	javaObject: Any?
+) : RawPojoDescriptor(mutability, javaObject) {
+	@AvailMethod
+	override fun o_Equals(self: AvailObject, another: A_BasicObject): Boolean =
+		another.equalsEqualityRawPojoFor(self, javaObject)
+
+	@AvailMethod
+	override fun o_EqualsEqualityRawPojo(
+		self: AvailObject,
+		otherEqualityRawPojo: AvailObject,
+		otherJavaObject: Any?
+	): Boolean = when {
+		javaObject == null -> otherJavaObject == null
+		otherJavaObject == null -> false
+		// Neither is null.
+		javaObject != otherJavaObject -> false
+		// They're equal.
+		!self.sameAddressAs(otherEqualityRawPojo) -> {
+			// They're equal, but distinct AvailObjects.
+			if (!isShared) {
+				self.becomeIndirectionTo(otherEqualityRawPojo)
+			} else if (!otherEqualityRawPojo.descriptor().isShared) {
+				otherEqualityRawPojo.becomeIndirectionTo(self)
+			}
+			true
+		}
+		else -> true
+	}
```

Comment:
We generally put opening braces for things like if, else, when on new lines. Maybe like:

```
): Boolean = 
	when 
	{
		javaObject == null -> otherJavaObject == null
		otherJavaObject == null -> false
		// Neither is null.
		javaObject != otherJavaObject -> false
		// They're equal.
		!self.sameAddressAs(otherEqualityRawPojo) -> {
			// They're equal, but distinct AvailObjects.
			if (!isShared)
		        {
				self.becomeIndirectionTo(otherEqualityRawPojo)
			} 
			else if (!otherEqualityRawPojo.descriptor().isShared) 
			{
				otherEqualityRawPojo.becomeIndirectionTo(self)
			}
			true
		}
		else -> true
	}
```

Reply:
I've updated the project's format settings to support this correctly.  Prior to this change, every time a "}" was typed, it would reformat the enclosed expression with the open brace on the same line.  Now it only does that for lambda arguments outside parentheses (because otherwise it's a syntax error).